### PR TITLE
feat(json-api-nestjs): Uuid support for relationships

### LINKED ISSUE #40: Full `uuid` support for primary keys
I've seen there's been progress towards `uuid` support but also noticed that relationships won't play nicely as it is right now.

I looked around a bit and one of the ideas I had is in the draft PR linked below.

Please let me know your opinions.

### PR DESCRIPTION
**Pull request by too-soon**
Fixes #40.

@klerick please take a look at my new approach based on your suggestion. Let me know if we can further improve it.

~Also, I've had a little trouble making an unit test because `pg-mem` doesn't seem to support `uuid` out of the box. Any ideas on how to solve that?~ (nvm, working on that)

### CODE REVIEW

File: libs/json-api-nestjs/src/lib/mixin/pipes/index.ts [hunk 1]
```diff
@@ -95,10 +96,6 @@ export function bodyRelationshipPatchPipeMixin(): PipeMixin {
...
 }
 
-export function idPipeMixin(
-  entity: Entity,
-  connectionName: string,
-  config: ConfigParam
-): PipeMixin {
-  return config.pipeForId;
+export function idPipeMixin(entity: Entity, connectionName: string): PipeMixin {
+  return factoryMixin(entity, connectionName, IdSchemaPipe);
```
Comment (klerick) [inline libs/json-api-nestjs/src/lib/mixin/pipes/index.ts]: I think will be better use config, by default use **ParseIntPipe**, by if need use other pipe, provide **ParseUUIDPipe**. 
```
import { ParseUUIDPipe } from '@nestjs/common';
....
JsonApiModule.forRoot({
....,
options: {
pipeForId: ParseUUIDPipe
}
})
```

What do you think?

**Comment by klerick** [comment]
> Also, I've had a little trouble making an unit test because `pg-mem` doesn't seem to support `uuid` out of the box. Any ideas on how to solve that?

if **pg-mem** does not support [type](https://github.com/oguimbal/pg-mem#custom-types) of [function](https://github.com/oguimbal/pg-mem#custom-functions) you can add it yourself.

File: libs/json-api-nestjs/src/lib/mixin/pipes/index.ts (hunk 1 above)
Comment (too-soon) [reply libs/json-api-nestjs/src/lib/mixin/pipes/index.ts]: That would be clean, indeed.

I only worry if there's some use-case to mix and match between `SEQ` and `UUID` at the same time. Like I'm testing right now with both generation strategies.

Personally I wouldn't mind if we make it a config parameter like you suggested, though. I'll give it a try over the weekend.

Comment (too-soon) [reply libs/json-api-nestjs/src/lib/mixin/pipes/index.ts]: I realize now it was already implemented. I will be using this method, no problem!

**Comment by too-soon** [comment]
I believe this should be good now, @klerick. Please take another look.

**Review by klerick — COMMENTED** [review]
LGTM

**Review by klerick — APPROVED** [review]

**Pull request merged**